app/controllers/concerns/course/lesson_plan/personalization_concern.rb

```diff
@@ -165,6 +167,29 @@ def algorithm_stragglers(course_user)
       each { |item| item.reload.find_or_create_personal_time_for(course_user).update(fixed: true) }
   end
 
+  # Returns cached data for the course user, if available, else it does the necessary computations and caches.
+  # Data returned is an array containing:
+  # - The submitted lesson plan item IDs
+  # - The published lesson plan items with reference and personal times, sorted by start_at for user
+  # - Learning rate computed for the user
+  # in the above order.
+  def retrieve_or_compute_course_user_data(course_user)
+    Rails.cache.fetch("course/lesson_plan/personalization_concern/#{course_user.id}", expires_in: 1.hours) do
```

Comment:
Any specific reason for the 1 hour cache?

Reply:
It's arbitrary. The cache currently only serves a single purpose, which is to eliminate a single redundant recomputation for people on OTOT. So the 1 hour is just there to make sure that between the two method calls (which shouldn't be spaced apart by > 1 hour), the data is still in the cache.

Between re-runs of the entire computation algorithm, the cache is cleared for the user, so no savings there.

Subsequent changes will try to make greater use of the caching.